### DISCUSSION TITLE
Is it possible to run hardhat in the browser?

### DISCUSSION THREAD

**Discussion by orenyomtov** [NomicFoundation/hardhat#3127] [Troubleshooting] [answered]
I tried naively running it through browserify, but I keep getting various errors.

I was wondering if someone else might have already gone through the process, or if there are some known reasons why it can't feasibly work.

**Comment by alcuadrado** [comment]
Hey, @orenyomtov!

Hardhat depends on tons of apis from Node.js and its dependency model, so it's not really browser-compatible.

Are you looking to run Hardhat or Hardhat Network in the browser?

**Comment by orenyomtov** [comment]
@alcuadrado thanks for the quick response!

Mainly Hardhat Network - does that make things easier?

**Comment by alcuadrado** [comment]
Thanks for clarifying. We have plans for that, but I don't think we'd be able to prioritize them this year.

**Comment by orenyomtov** [comment]
Do you have a general plan of attack for supporting Hardhat Network to run in the browser?
Maybe I'll be able to lend a hand with some of the work

**Comment by alcuadrado** [comment, marked as answer]
We are porting it to rust and will provide wasm builds.

**Comment by orenyomtov** [comment]
Oh god wow

Well good luck with that 😅

![image](https://user-images.githubusercontent.com/168856/188328672-f0620e67-2acc-40da-bd24-19a07f554b76.png)
